contracts/TiramisuSavingsClub.sol

```diff
@@ -24,19 +24,22 @@ contract TiramisuSavingsClub {
     Group[] public groups;
 
     mapping(address => uint) public memberToGroupId;
-    mapping(address => uint) private deposits;
-    mapping(address => uint) private withdrawals;
+    mapping(address => uint) deposits;
+    mapping(address => uint) withdrawals;
+    mapping(address => string) memberNames;
 
-    event Deposit(address depositor, uint groupId, uint amount);
-    event Withdrawal(address payee, uint groupId, uint amount);
+    event NewGroupEvent(address[] addresses, string[] names, uint ownerIndex);
```

Comment:
Can we change NewGroupEvent to simply include the group id? Just a small gas optimization

Front end might then need to explicitly fetch group information for that id

Reply:
Actually I have the same question on all of these events.

Is it generally best practice to be verbose and redundant with event data?
Or is it better to provide the minimal important data and let the frontend query the real state instead of relying on events?

I imagine there could be security implications for the frontend relying on events instead of the state explicitly.

Reply:
Ok if you follow along on my Discord thread, Sanjay helped answer the question.
Basically it's a tradeoff and I think this is ok without further changes.

I think it's better in our case to have verbose events (even if they are redundant with the state). This could allow us to rely less on public view functions (which also are ok)

Closing this comment